src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1702,6 +1702,10 @@ theorem allOnes_shiftLeft_or_shiftLeft {x : BitVec w} {n : Nat} :
     BitVec.allOnes w <<< n ||| x <<< n = BitVec.allOnes w <<< n := by
   simp [← shiftLeft_or_distrib]
 
+@[simp] theorem setWidth_shiftLeft_of_le (hi : i ≤ v) (a : BitVec v) (b : Nat) :
+    (a <<< b).setWidth i = a.setWidth i <<< b :=
+  eq_of_getElem_eq (fun j hj => Bool.eq_iff_iff.2 (by simp; omega))
```

Comment:
```suggestion
@[simp] theorem setWidth_shiftLeft_of_le {x : BitVec w} {y : Nat} (hi : i ≤ w)  :
    (x <<< y).setWidth i = x.setWidth i <<< y :=
  eq_of_getElem_eq (fun j hj => Bool.eq_iff_iff.2 (by simp; omega))
```

Again, just for your info, how would I have named things based on our previous conventions? Not a blocker.